src/cli/rustup_mode.rs

```diff
@@ -169,6 +169,12 @@ pub fn cli() -> App<'static, 'static> {
                         .required(true)
                         .multiple(true),
                 )
+                .arg(
+                    Arg::with_name("profile")
+                        .long("profile")
+                        .possible_values(Profile::names())
+                        .default_value(Profile::default_name()),
+                )
```

Comment:
While this looks good, Sadly this is only for the "hidden" `rustup install` command.  A little later in the file there's a `toolchain` subcommand which has the real `rustup toolchain install` command which you'll also need to add this flag to.